### ISSUE TITLE
[Phase 1] Data Infrastructure - Sequence Loading

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/protein-structure-prediction#2]
## Objective

Build the data loading infrastructure for protein secondary structure datasets including CB513 and CullPDB.

**Part of:** #1

## Dependencies

None (first phase)

### Data Loader Implementation
- [ ] Create `src/data/loader.py` with `ProteinDataLoader` class
- [ ] Support CB513 and CullPDB dataset formats
- [ ] Implement 8-state to 3-state label reduction

### Sequence Parser
- [ ] Create `src/data/parser.py` for FASTA parsing
- [ ] Handle variable-length sequences
- [ ] Validate amino acid sequences

### Data Validator
- [ ] Create `src/data/validator.py`
- [ ] Check sequence-label length matching
- [ ] Validate amino acid alphabet

### ProteinDataLoader Skeleton

```python
# src/data/loader.py
from typing import List, Tuple

DSSP_TO_Q3 = {
    'H': 'H', 'G': 'H', 'I': 'H',
    'E': 'E', 'B': 'E',
    'T': 'C', 'S': 'C', 'C': 'C', '-': 'C'
}

class ProteinDataLoader:
    def __init__(self, dataset: str, data_dir: str = "data/raw"):
        self.dataset = dataset.lower()
        self.data_dir = data_dir

    def load(self) -> Tuple[List[str], List[str]]:
        """Load sequences and structure labels."""
        pass

    def reduce_labels(self, labels_8state: str) -> str:
        """Reduce 8-state to 3-state labels."""
        return ''.join(DSSP_TO_Q3.get(c, 'C') for c in labels_8state)
```

### First Test

```python
# tests/unit/test_loader.py
import pytest
from src.data.loader import ProteinDataLoader, DSSP_TO_Q3

def test_label_reduction():
    loader = ProteinDataLoader("cb513")
    assert loader.reduce_labels("HHHEEECCC") == "HHHEEECC"
    assert loader.reduce_labels("GGGBBBSSS") == "HHHEECCCC"
```

## Definition of Done

- [ ] Can load CB513 dataset from numpy format
- [ ] Label reduction produces valid Q3 labels
- [ ] Sequence validation catches invalid amino acids
- [ ] Unit tests pass with >80% coverage

## Technical Notes

- CB513 uses Princeton's numpy format with profile features
- Each protein is padded to 700 residues
- First 21 columns are one-hot amino acids, next 21 are PSSM
